### PR TITLE
HPCC-28811 Remove configuration logic related to SPARK-HPCC

### PR DESCRIPTION
**Pull request by kenrowland**
Removed config files and other references to SPARK configuration

Signed-Off-By: Kenneth Rowland kenneth.rowland@lexisnexisrisk.com

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [x] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by rpastrana — COMMENTED** [review]
@kenrowland Michael Gardner is a better reviewer for this change. Also, make sure this is not merged until the feature is removed.

**Comment by kenrowland** [comment]
Michael added to review list

File: initfiles/etc/DIR_NAME/environment.xml.in [hunk 1]
```diff
@@ -969,25 +969,6 @@
              name="s1"
              netAddress="."/>
   </TopoServerProcess>
-  @SPARK_ENV_S@
```
Comment (Michael-Gardner) [inline initfiles/etc/DIR_NAME/environment.xml.in:972]: This variable is coming from initfiles/etc/DIR_NAME/CMakeLists.txt around line 20-26?

Should remove that block as part of this PR. 

Also remove @DNG_SPARKTHOR@ from initfiles/etc/DIR_NAME/genenvrules.conf.in at the end of line 4

**Review by Michael-Gardner — CHANGES_REQUESTED** [review]
see comment about environment.xml.in variable

**Comment by kenrowland** [comment]
@Michael-Gardner How about the spark reference in initfiles/bash/etc/systemd/system/thorslave@instance.service.in,
or is that beyond the scope of this Jira?

**Comment by Michael-Gardner** [comment]
@kenrowland I think that's beyond the scope. I'll go ahead and clean up files like that after your PR is done. Lets just keep your PR targeted towards the environment and configuration files.

File: initfiles/etc/DIR_NAME/environment.xml.in (hunk 1 above)
Comment (Michael-Gardner) [reply initfiles/etc/DIR_NAME/environment.xml.in:972]: Perfect thanks.

**Review by Michael-Gardner — APPROVED** [review]
Looks great now. Thanks for the change @kenrowland

**Comment by kenrowland** [comment]
@ghalliday Please merge

**Review by rpastrana — COMMENTED** [review]
@kenrowland I have no input on this change, @Michael-Gardner is the expert in this area and his approval should suffice

**Comment by kenrowland** [comment]
@ghalliday Not sure how to officially mark it as such, but this PR should only be merged when the feature is fully removed. There is no more work for this PR and it is ready for merge when the rest of the work is complete.

**Comment by ghalliday** [comment]
@kenrowland please squash.  Also, what is stopping this from being merged?  If it is dependent on something else what are the jira numbers?

**Comment by kenrowland** [comment]
@ghalliday rebased. This is a sub-task of HPCC-28520 "Sunset integrated SPARK feature". Per comments from @rpastrana earlier in this PR, these changes should not  be merged until the feature is removed. There are still several open sub tasks on the parent Jira, so I wanted to make sure it wasn't merged before it should be.

**Pull request merged**